lib/MPIPreferences/src/MPIPreferences.jl

```diff
@@ -0,0 +1,181 @@
+module MPIPreferences
+
+using Preferences, Libdl
+
+const binary = @load_preference("binary", Sys.iswindows() ? "MicrosoftMPI_jll" : "MPICH_jll")
```

Comment:
I wasn't sure if this should be a function or a `const` binding.

Reply:
I think a const binding is fine, and will trigger recompilation if it changes.